src/Query.php

```diff
@@ -110,7 +108,7 @@ public function reset()
             'filter'         => null,
             'facet'          => null,
             'subfilters'     => null,
-            'weightedsearch' => null,
+            'weightedsearch' => null, // first getWeightedSearch() call will set this to true or false
```

Comment:
@j3nsch Soweit ich es verstehe, muss der initiale Wert für `'weightedsearch'` im Datenarray `null` sein. Wenn er initial bereits `false` wäre, so würde immer das `search.weightedSearch` Setting in der Konfiguration gewinnen: Falls die gewichtete Suche in der Konfiguration eingeschaltet wurde, könnte man sie in einem Test nicht mehr direkt über `$query->setWeightedSearch(false);` ausschalten.

Reply:
Der `getWeightedSearch()` Getter wird beim 1. Aufruf auf jeden Fall entweder `true` oder `false` im Datenarray setzen.

Die expliziten Getter haben gegenüber der vorherigen Implementierung auch den Vorteil, dass die Konfiguration stets nur einmal ausgelesen wird.

Reply:
Wir habe beschlossen `getWeightedSearch` direkt zu implementieren und nicht dem Magic-Get zu überlassen.

Reply:
Ok, das werde ich so umsetzen.